### PR TITLE
Add company-shell to shell-scripts layer

### PR DESCRIPTION
**Pull request by joehillen**

### CODE REVIEW

File: layers/+lang/shell-scripts/packages.el [hunk 1]
```diff
@@ -41,3 +43,18 @@
...
+    (spacemacs|add-company-hook fish-mode))
+
+  (defun shell-scripts/init-company-shell ()
+    (use-package company-shell
+      :if (configuration-layer/package-usedp 'company)
+      :defer t
+      :init
+      (push 'company-shell      company-backends-sh-mode)
+      (push 'company-shell      company-backends-fish-mode)
+      (push 'company-fish-shell company-backends-fish-mode))))
```
Comment (fbergroth) [inline layers/+lang/shell-scripts/packages.el]: In company, only one backend is used at a time. You probably want
```el
(push '(company-shell company-fish-shell) company-backends-fish-mode)
```
i.e. a grouped backend.

Also wrap the the setup code in a `progn`.

Comment (joehillen) [inline layers/+lang/shell-scripts/packages.el]: Fixed and squashed. Thank you!

**Comment by CestDiego** [comment]
YAS

**Comment by syl20bnr** [comment]
Nice :purple_heart: 

Thank you :+1: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**